ext/testing/promise_tracker.rs

```diff
@@ -18,6 +23,28 @@ impl PromiseMetricsSummary {
     pub fn has_pending_promises(&self) -> bool {
         self.promises_initialized > self.promises_resolved
     }
+
+    pub fn contains(&self, promise_id: NonZeroI32) -> bool {
+        self.promises.contains(&promise_id)
+    }
+
+    pub fn initialized(&mut self, promise_id: NonZeroI32) {
+        self.promises_initialized += 1;
+        debug_assert!(
+            self.promises.insert(promise_id),
```

Comment:
I'm pretty sure this will only run `self.promises.insert(promise_id)` in debug mode. (Same for `resolved`)